elephant/change_point_detection.py

```diff
@@ -0,0 +1,514 @@
+        # automatic setting of dt
+        if dt_temp is None:  
+            dt_temp = h / 20.
+        # filter_process for window of size h
+        t, differences = _filter_process(dt_temp, h, spk, t_final, test_param)
+        time_index = np.arange(len(differences))
+        # Point detected with window h
+        cps_window = []
+        while np.max(differences) > test_quantile:
+            cp_index = np.argmax(differences)
+            # from index to time
+            cp = cp_index * dt_temp + h  
+            #print("detected point {0}".format(cp), "with filter {0}".format(h))
+            # before to repet the procedure the h-neighbourg of 'cp' detected 
+            # is cut, because rate changes within it are explained by this cp
+            differences[np.where(
+                (time_index > cp_index - int(h / dt_temp.rescale(h.units))) & (
+                        time_index < cp_index + int(
+                            h / dt_temp.rescale(h.units))))[0]] = 0
+            # check if the neighbourhood of detected cp does not contain cps 
+            # detected with other windows
+            neighbourhood_free = True
+            if i == 0:
+                cps_window.append(cp)
+            else:
+                # iterate on lists of cps detected with smaller window
+                for j in range(i):
+                    # iterating on cps detected with the j^th smallest window
+                    for c_pre in cps[j]:
+                        if c_pre - h < cp < c_pre + h:
+                            neighbourhood_free = False
+                            break
+                # if none of the previous detected cp falls in the h
+                # neighbourhood
+                if neighbourhood_free:
+                    # add the point to the list
+                    cps_window.append(cp) 
+        # add the list to the list
+        cps.append(cps_window)  
+
+    return cps
+
+
+def _brownian_motion(t_in, t_fin, x_in, dt):
+    """
+    Generate a Brownian Motion.
+
+    Parameter
+    ---------
+        t_in : quantities,
+            initial time
+        t_fin : quantities,
+             final time
+        x_in : quantities,
+            initial point of the process: _brownian_motio(0) = x_in
+        dt : quantities,
+          resolution, time step at which brownian increments are summed
+    Returns
+    -------
+    Brownian motion on [t_in, t_fin], with resolution dt and initial state x_in
+    """
+
+    u = 1 * pq.s
+    try:
+        t_in_sec = t_in.rescale(u)
+    except ValueError:
+        raise ValueError("t_in must be a time quantity")
+    t_in_m = t_in_sec.magnitude
+    try:
+        t_fin_sec = t_fin.rescale(u)
+    except ValueError:
+        raise ValueError("t_fin must be a time quantity")
+    t_fin_m = t_fin_sec.magnitude
+    try:
+        dt_sec = dt.rescale(u)
+    except ValueError:
+        raise ValueError("dt must be a time quantity")
+    t_fin_m = t_fin_sec.magnitude
+
```

Comment:
`t_in_sec` and `t_fin_sec` are not actually used.
You should rather define them as `t_in_sec = t_in.rescale(u).magnitude` and `t_fin_sec = t_fin.rescale(u).magnitude`, and use them instead of `t_in_m` and `t_fin_m`.
This kind of confusing usage of rescale is seen in other functions as well.
Please correct them as well so that the treatment of units is consistent within the code.